e2e/require-main-after-create-require/__tests__/parent.test.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const Module = require('module');
+const path = require('path');
+
+test('require child from parent', () => {
+  // createRequire with a different file
+  require = Module.createRequire(path.resolve('./empty.js'));
+  require('./child');
+
+  expect(1).toBe(1);
```

Comment:
idea to make this nicer to read:
```
expect(() => require('./child')).not.toThrow()
```
and in `child.js`
```
process.stdout.write(require.main.filename)
```
which would throw if `require.main` is `null`
Hope this makes sense and seems like a good idea not just to me :D

Reply:
Thanks, this is genius!